src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -81,10 +81,21 @@ public async Task RunApplicationAsync(CancellationToken cancellationToken = defa
         AspireEventSource.Instance.DcpModelCreationStart();
         try
         {
-            if (!_model.Resources.Any(r => r.Name == DashboardReservedResourceName))
+            var dashboardIndex = _model.Resources.TakeWhile(static r => !StringComparers.ResourceName.Equals(r.Name, KnownResourceNames.AspireDashboard)).Count();
+
+            if (dashboardIndex is -1)
             {
+                // No dashboard is specified, so start one.
+                // TODO validate that the dashboard has not been suppressed
                 await StartDashboardAsync(cancellationToken).ConfigureAwait(false);
             }
+            else if (dashboardIndex is not 0)
+            {
+                // A dashboard exists in the set. Move it to the front so it starts first.
+                var dashboardResource = _model.Resources[dashboardIndex];
+                _model.Resources.RemoveAt(dashboardIndex);
+                _model.Resources.Insert(0, dashboardResource);
+            }
```

Comment:
This is the key change here. If the dashboard is not first, move it to the first position.